packages/core/src/util/txInspector.ts

```diff
@@ -24,35 +28,96 @@ export type StakeKeyRegistrationInspection = StakeAddressCertificate[];
 export type WithdrawalInspection = Lovelace;
 
 // Inspector types
+interface SentInspectorArgs {
+  addresses?: Address[];
+  stakeKeyHashes?: Ed25519KeyHash[];
+  rewardAccounts?: RewardAccount[];
+}
+export type SentInspector = (args: SentInspectorArgs) => Inspector<boolean>;
+export type TotalInputValuesInspector = (
+  ownAddresses: Address[],
+  historicalTxs: TxAlonzo[]
```

Comment:
This would allow using the same inspector object for a longer period of time
```suggestion
  getHistoricalTxs: () => TxAlonzo[]
```